projects/wave-core/src/lib/operators/dialogs/data-repository/vector/vector-source-dataset.component.ts

```diff
@@ -96,6 +96,68 @@ export class VectorSourceDatasetComponent implements OnInit {
+                        let globalDatasetMax = this.dataset.time_end;
+                        this.dataset.vectorLayer.forEach((element) => {
+                            for (const sourceVar in element) {
+                                if (element.hasOwnProperty(sourceVar)) {
+                                    if (sourceVar === 'time_start') {
+                                        if (element.time_start.isValid()) {
+                                            if (!this._displayedColumns.includes('start')) {
+                                                this._displayedColumns.push('start');
+                                            }
+                                            if (element.time_start < globalDatasetMin) {
+                                                globalDatasetMin = element.time_start;
+                                            }
+                                        }
+                                    }
+                                    if (sourceVar === 'time_end') {
+                                        if (element.time_end.isValid()) {
+                                            if (!this._displayedColumns.includes('end')) {
+                                                this._displayedColumns.push('end');
+                                            }
+                                            if (element.time_end > globalDatasetMax) {
+                                                globalDatasetMax = element.time_end;
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        });
+                        // Fill in the global Min/Max for rows without Start/End
+                        this.dataset.vectorLayer.forEach((element) => {
+                            for (const sourceVar in element) {
+                                if (element.hasOwnProperty(sourceVar)) {
+                                    if (sourceVar === 'time_start') {
+                                        if (!element.time_start.isValid()) {
+                                            element.time_start = globalDatasetMin;
+                                        }
+                                    }
+                                    if (sourceVar === 'time_end') {
+                                        if (!element.time_end.isValid()) {
+                                            element.time_end = globalDatasetMax;
+                                        }
+                                    }
+                                }
+                            }
+                        });
+                        // Update the global Min/Max in the whole dataset
+                        this.dataset.time_start = globalDatasetMin;
+                        this.dataset.time_end = globalDatasetMax;
```

Comment:
If you change the dataset, then move this processing to the `UserService`.

Reply:
reverted the dataset manipulation/processing part in bcf99b0df6ed19d67ce802b25df192cf7edd124e

As discussed i would like to do the dataset processing part in a new PR